static/js/settings.js

```diff
@@ -414,6 +419,23 @@ var SettingsScreen = {
     });
   },
 
+  showAddonConfigScreen: function(id) {
+    this.backButton.href = '/settings/addons';
+    this.addonMainSettings.classList.add('hidden');
+    this.addonConfigSettings.classList.remove('hidden');
+
+    this.fetchAddonList().then(() => {
```

Comment:
You don't necessarily need to fetch the add-on list here. Instead, you can make sure it's already been fetched, like `showDiscoveredAddonsScreen` does:

```javascript
     let promise;
     if (this.installedAddons.size === 0 && this.availableAddons.size === 0) {
       promise = this.fetchAddonList();
     } else {
       promise = Promise.resolve();
     }

     promise.then(() => {
```